lib/spack/spack/cmd/diff.py

```diff
@@ -90,16 +78,16 @@ def compare_specs(a, b, to_string=False, colorful=True):
 
     # Format the spec names to be colored
     fmt = "{name}{@version}{/hash}"
-    a_name = a.format(fmt, color=color.get_color_when())
-    b_name = b.format(fmt, color=color.get_color_when())
+    a_name = a.format(fmt, color=color)
+    b_name = b.format(fmt, color=color)
 
     # We want to show what is the same, and then difference for each
     return {
         "intersect": flatten(intersect) if to_string else intersect,
         "a_not_b": flatten(spec1_not_spec2) if to_string else spec1_not_spec2,
         "b_not_a": flatten(spec2_not_spec1) if to_string else spec2_not_spec1,
-        "a_name": a_name if colorful else a.format("{name}{@version}{/hash}"),
-        "b_name": b_name if colorful else b.format("{name}{@version}{/hash}")
+        "a_name": a_name,
```

Comment:
Can you show me what this looks like for both json and standard output in color?

Reply:
<img width="528" alt="Screen Shot 2021-08-05 at 10 18 04 PM" src="https://user-images.githubusercontent.com/299842/128459758-f7b858a5-fb79-4996-a9bf-750c1181c59a.png">
<img width="380" alt="Screen Shot 2021-08-05 at 10 18 19 PM" src="https://user-images.githubusercontent.com/299842/128459764-36ef88c5-db28-4b7c-b6e0-b21c22a932b7.png">

Reply:
That looks great!